File: src/Backend.Core/Models/Tag.cs
namespace Backend.Models;

public class Tag
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int UserId { get; set; }

    public override bool Equals(object? other)
    {
        if (other is not Tag otherTag)
        {
            return false;
        }
        return Id == otherTag.Id 
               && Name == otherTag.Name 
               && UserId == otherTag.UserId;
    }
}

File: src/webfrontend/src/models/Tag.ts
import { Comparable } from "./Comparable"

export class Tag implements Comparable<Tag> {
    public id: number = 0
    public name: string = ""

    constructor(id: number, name: string) {
        this.id = id
        this.name = name
    }
    Equals(other: Tag): boolean {
        return this.id === other.id
            && this.name === other.name
    }
}